### PR TITLE
CMOS-169: Internal log ingestion of our own logs into Loki

### PR DESCRIPTION
**Pull request by patrick-stephens**
Using Promtail because Fluent Bit is not supported for Alpine and seems to trigger segmentation faults when sending to Loki.
Updated `cmosinfo` script as well.
Added licensing information too.

Screenshot 2021-11-19 at 17 04 19

cc: @ingenthr - logs are ingested but not on any default dashboard yet, screenshot is from the explore tab.

### CODE REVIEW

File: microlith/Dockerfile [hunk 1]
```diff
@@ -254,6 +274,8 @@ ENV GF_INSTALL_PLUGINS=""
 COPY prometheus/ /etc/prometheus/
 COPY alertmanager/ /etc/alertmanager/
 COPY jaeger/ /etc/jaeger/
+COPY promtail/ /etc/promtail/
+ENV CMOS_LOGS=/logs
```
Comment (markspolakovs) [inline microlith/Dockerfile:278]: NITPICKING: possibly `CMOS_LOGS_ROOT` or a more descriptive name?

File: microlith/run.sh [hunk 2]
```diff
@@ -76,6 +81,15 @@ if [[ -v "KUBERNETES_DEPLOYMENT" ]]; then
     log "Using Kubernetes mode as KUBERNETES_DEPLOYMENT set (value ignored)"
 fi
 
+if [[ ! -x /bin/cbmultimanager ]]; then
+    log "Running OSS version, no Couchbase binaries"
+    export CMOS_OSS_VERSION=true
+    export CMOS_VERSION="OSS"
```
Comment (markspolakovs) [inline microlith/run.sh:87]: Hm, `VERSION` implies a version number - perhaps `DISTRIBUTION` or `FLAVOR`?

**Review by markspolakovs — COMMENTED** [review]
Looks sensible, though I'm not a huge fan of needing to bring in glibc and systemd just for self-log tailing - though if there's no better way it'll have to do.

Mind posting a screenshot of the new dashboard?

File: microlith/run.sh (hunk 2 above)
Comment (patrick-stephens) [reply microlith/run.sh:87]: DISTRIBUTION rather than naughty spelling!

File: microlith/Dockerfile (hunk 1 above)
Comment (markspolakovs) [reply microlith/Dockerfile:278]: Also, if we're making this user-customisable, should we update the `tee` in `run.sh` to use that variable instead? Currently it's hardcoded to write to `/logs/"${EXE_NAME}".log`.

Comment (patrick-stephens) [reply microlith/Dockerfile:278]: Yeah I did but then the rebase overwrote it!

**Review by markspolakovs — APPROVED** [review]

**Pull request merged**